pkg/aggregator/aggregator.go

```diff
@@ -875,6 +883,9 @@ func (agg *BufferedAggregator) tags(withVersion bool) []string {
 		if version.AgentPackageVersion != "" {
 			tags = append(tags, "package_version:"+version.AgentPackageVersion)
 		}
+		if agg.configID != "" {
+			tags = append(tags, "config_id:"+agg.configID)
+		}
```

Comment:
@coignetp Is there a specific reason to put this under `if withVersion {` condition?

For HA, we need `config_id` too, but we don't have it now due to this condition, we are using `agg.tags(false)`

https://github.com/DataDog/datadog-agent/blob/3c0593ceeb4c2e810eebc43751735f7f95ec58f4/pkg/aggregator/aggregator.go#L621-L624

Reply:
Alternatively, I can use `agg.tags(true)` for `datadog.agent.ha_agent.running` metric.

Reply:
The main reason is to only attach it to `datadog.agent.running` metric, and `configID` is technically a version on our side (tho that's a detail)

Reply:
Got it, I will update `datadog.agent.ha_agent.running ` to use `agg.tags(true)` then.

Reply:
The tag will indeed only be added if there is a `config_id` value.
It's added to the tag list of a metric when `agg.tags(true)` is used, and `datadog.agent.running` ([code](https://github.com/DataDog/datadog-agent/blob/b6b1c510e74e40e6161b8f11a42c5f3291376c6f/pkg/aggregator/aggregator.go#L606)) and `datadog.agent.started` ([code](https://github.com/DataDog/datadog-agent/blob/b6b1c510e74e40e6161b8f11a42c5f3291376c6f/pkg/aggregator/demultiplexer_agent.go#L244)) - emitted once at the start of an agent -- are the only one using it (the other use `agg.tags(false)`)
Alexandre may want to add the tag too, and use `agg.tags(true)` as well

Reply:
Gotcha, I see. It is minor, but it feels a bit awkward to me to overload `withVersion` to also add this tag. Do we always want `config_id` whenever we add `version`? Naively, I'd have expected us to add this tag to the code you linked where we create the `datadog.agent.running` metric if we only need it there.

Reply:
> Naively, I'd have expected us to add this tag to the code you linked where we create the datadog.agent.running metric if we only need it there.

+1, I think that makes sense.

Reply:
@coignetp I know this was already merged, but what do you think about making this refactoring change? Or are there reasons you think the current implementation is preferable?